File: futurespro-backend/app/schemas/strategy.py
from pydantic import BaseModel
from typing import Optional, Dict, Any
from decimal import Decimal
from datetime import datetime


class StrategyBase(BaseModel):
    """策略基础模式"""
    name: str
    description: Optional[str] = None
    type: str
    config: Optional[Dict[str, Any]] = None


class StrategyCreate(StrategyBase):
    """策略创建模式"""
    pass


class Strategy(StrategyBase):
    """策略模式"""
    id: int
    user_id: int
    status: str
    total_pnl: Decimal
    win_rate: Optional[Decimal] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

File: futurespro-frontend/src/types/strategy.ts
export interface Strategy {
  id: number;
  user_id: number;
  name: string;
  description: string;
  type: string;
  config: Record<string, any>;
  status: string;
  total_pnl: number;
  win_rate: number | null;
  created_at: string;
  performance?: {
    total_return: number;
    win_rate: number;
    max_drawdown: number;
  };
}

export interface StrategyCreate {
  name: string;
  description: string;
  type: string;
  config: Record<string, any>;
}

export interface StrategyUpdate {
  name: string;
  description: string;
  type: string;
  config: Record<string, any>;
}
